extensions/apps/settings/src/components/pages/notifications-preferences-option.tsx

```diff
@@ -48,9 +174,173 @@ const NotificationsPreferencesOption: React.FC = () => {
+        )}
+      </Text>
+      <Card padding="p-3" customStyle="mt-4" background={{ light: 'grey9', dark: 'grey3' }}>
+        <Stack direction="row" spacing="gap-x-3">
+          {isDarkTheme ? <InfoLight className="shrink-0" /> : <InfoDark className="shrink-0" />}
+          <Text variant="body1" customStyle="text-sm">
+            {t('Changing notifications preferences requires a signature')}
+          </Text>
+        </Stack>
+      </Card>
+    </Stack>
+  );
+};
+
+const ProfileSetting = ({ isSelected, onChange }) => {
+  const { t } = useTranslation('app-settings-ewa');
+
+  return (
+    <Stack customStyle="border(b-1 solid grey8 dark:grey5) mb-4 pb-4">
+      <Stack direction="row" justify="between" align="center" customStyle="mt-4">
+        <Text variant="body1">{t('Profile')}</Text>
+        <Checkbox
+          id="profile-checkbox"
+          value="Profile"
+          name="profile"
+          isSelected={isSelected}
+          handleChange={onChange}
+          size="large"
+          customStyle="w-6 h-6"
+        />
+      </Stack>
+
+      <Text variant="footnotes2" weight="normal" customStyle="dark:text-grey6 text-grey4 mt-2">
+        {t('Get notifications about new followers')}
+      </Text>
+    </Stack>
+  );
+};
+
+const EnableAllSetting = ({ isSelected, onChange }) => {
+  const { t } = useTranslation('app-settings-ewa');
+
+  return (
+    <Stack customStyle="border(b-1 solid grey8 dark:grey5) mb-4">
+      <Stack direction="row" justify="between" align="center" customStyle="my-4">
+        <Text variant="body1">{t('Enable all')}</Text>
+        <Checkbox
+          id="enable-all-notifications-checkbox"
+          value="Enable all"
+          name="enable-all"
+          isSelected={isSelected}
+          handleChange={onChange}
+          size="large"
+          customStyle="w-6 h-6"
+        />
+      </Stack>
+    </Stack>
+  );
+};
+
+const UnlockCard = ({ isDarkTheme, loading, onClick }) => {
+  const { t } = useTranslation('app-settings-ewa');
+
+  return (
+    <Card background={{ light: 'grey9', dark: 'grey3' }} padding="p-3">
+      <Stack direction="row" spacing="gap-x-3">
+        {isDarkTheme ? <LockLight /> : <LockDark />}
+        <Stack direction="column" spacing="gap-y-1">
+          <Text variant="button-md" color={{ dark: 'white', light: 'black' }}>
+            {t('Unlock preferences')}
+          </Text>
+          <Text variant="body2" color={{ dark: 'white', light: 'black' }}>
+            {t('Click “Unlock” to unlock preferences. You will be prompted with 1 signature.')}
+          </Text>
+          {
+            <Button
+              onClick={onClick}
+              variant="text"
+              size="md"
+              color="dark:secondaryLight secondaryDark"
+              label={t('Unlock')}
+              customStyle="mr-auto"
+              loading={loading}
+            />
+          }
+        </Stack>
+      </Stack>
+    </Card>
```

Comment:
`notifications-preferences-option` can be a folder, and the page can go inside `index.tsx` file.  And all the components I highlighted can be extracted and moved to their respective component file.

Reply:
And each of the component you extract can be properly typed.

Reply:
done 👍